### ISSUE TITLE
Cannot provide path aliases to vanilla-extract plugin integration

### ISSUE THREAD

**Issue by chinanderm** [remix-run/remix#7077]
### What version of Remix are you using?

1.19.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

In an Nx monorepo we are using vanilla-extract to style components that are imported by a handful of Remix applications. When a Remix application has `*.css.ts` files directly, or imports one via a shared lib/package in the monorepo with a _relative import_, everything works great. However, if the import is via a path alias (defined in tsconfig), Vite in the vanilla-extract plugin/integration process cannot resolve the path of the import. Furthermore, there is no ability to tell Remix about path aliases so it can pass those on to the vanilla-extract integration.

We've created [a reproduction repo](https://github.com/chinanderm/remix-vanilla-extract-vite-alias) that can highlight the situation.

After some debugging and an investigation, we've found where in vanilla-extract these aliases would need to be introduced: the call to `vite.createServer()` in [packages/integration/src/compiler.ts](https://github.com/vanilla-extract-css/vanilla-extract/blob/40d7e72c041989a4dc847dc66b94a4c680b5536e/packages/integration/src/compiler.ts#L61). Here, there is no ability to pass in a `resolve.alias` Vite configuration to define aliases. On the Remix side, there's no ability to define these aliases to pass through to vanilla-extract.

In the mean time, we've created a patch in our repo that adds in `resolve.alias` to the config object passed to `vite.createServer()`. This patch solves the problem for us in the short term but ideally Remix and/or the integration code in vanilla-extract can support an alias configuration, whether automatic or manual.

Seeing as this would likely need to be an update in both Remix and vanilla-extract, [an issue has also been opened](https://github.com/vanilla-extract-css/vanilla-extract/issues/1156) in that repo.

### Expected Behavior

The expected behavior is that path aliases can be resolved when processing `*.css.ts` files.

### Actual Behavior

Code imported in `*.css.ts` files via path aliases cannot be resolved, leading to build errors.

**Comment by MichaelDeBoey** [comment]
@chinanderm off topic: what generator did you use for the NX monorepo?

**Comment by chinanderm** [comment]
@MichaelDeBoey No generator, just a manual installation.

**Comment by MichaelDeBoey** [comment]
Anything special you did for recognizing dependencies as they're now in the root of the monorepo?

You can contact me on Discord btw if that makes it easier to chat about

**Comment by chinanderm** [comment]
@MichaelDeBoey Nope, nothing special. My hunch is that the Vite bit of the VE plugin/integration will first try to resolve from defined aliases, then local paths, and then finally node_modules. After that it gives up. Since there are no defined aliases, it cannot find our lib.

I hope that answers your question! Happy to provide more context but I think the reproduction repo sums it up succinctly without a lot of extraneous stuff.

Thanks to the team for taking a look at this! It's been one of the last remaining roadblocks in our code transition to Remix with Shopify/Hydrogen.

**Comment by Cjstoney** [comment]
Any update on this? We are facing the same issue with `remix:"^2.8.1"`  and we're using the vite plugin.

**Comment by chinanderm** [comment]
@Cjstoney My assumption is the answer is: "migrate to using Vite with Remix".

**Comment by Cjstoney** [comment]
@markdalgleish Just wanting to check if any progress has been made towards this?

**Comment by brophdawg11** [comment]
Thank you for opening this issue, and our apologies we haven't gotten around to it yet!  

With the release of [React Router v7](https://remix.run/blog/react-router-v7) we are sun-setting continued development/maintenance on Remix v2.  If you have not already [upgraded to React Router v7](https://reactrouter.com/upgrading/remix), we recommend you do so. We've tried to make the upgrade process as smooth as possible with our [Future Flags](https://remix.run/docs/en/main/start/future-flags). We are now in the process of cleaning up outdated issues and pull requests to improve the overall hygiene of our repositories.  

We plan to continue to address 2 types of issues in Remix v2:

* Bugs that pose security concerns
* Bugs that prevent upgrading to React Router v7

If you believe this issue meets one of those criteria, please respond or [create a new issue](https://github.com/remix-run/remix/issues/new?template=bug_report.yml).

For all other issues, ongoing maintenance will be happening in React Router v7, so:

* If this is a bug, please reopen this issue [in that repo](https://github.com/remix-run/react-router/issues/new?template=bug_report.yml) with a new [minimal reproduction](https://reactrouter.com/new) against v7
* If this is a feature request, please open a new [Proposal Discussion](https://github.com/remix-run/react-router/discussions/new?category=proposals) in React Router, and if it gets enough community support it can be considered for implementation

If you have any questions you can always reach out on [Discord](https://rmx.as/discord). Thanks again for providing feedback and helping us make our framework even better!
